### PR TITLE
do not add previous hosts predicate when consistent hashing is enabled

### PR DESCRIPTION
**Pull request by ramaraochavali**
Fixes https://github.com/istio/istio/issues/50756

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/networking/core/route/route.go [hunk 1]
```diff
@@ -622,8 +615,16 @@ func applyHTTPRouteDestination(
...
+	policy := in.Retries
+	if policy == nil {
+		// No VS policy set, use mesh defaults
+		policy = mesh.GetDefaultHttpRetryPolicy()
+	}
+	consistentHash := false
 	if len(in.Route) == 1 {
 		hostnames = append(hostnames, processDestination(in.Route[0], serviceRegistry, listenerPort, hashByDestination, out, action))
+		hash := hashByDestination[in.Route[0]]
+		consistentHash = hash != nil
```
Comment (hzxuzhonghu) [inline pilot/pkg/networking/core/route/route.go:627]: How do you handle consitent hash for the below else branch

Comment (ramaraochavali) [reply pilot/pkg/networking/core/route/route.go:627]: consistent hash weighted desitnations do not work together

File: pilot/pkg/networking/core/route/retry/retry.go [hunk 2]
```diff
@@ -73,7 +90,6 @@ func ConvertPolicy(in *networking.HTTPRetry) *route.RetryPolicy {
 	}
 
 	// A policy was specified. Start with the default and override with user-provided fields where appropriate.
-	out := DefaultPolicy()
```
[the remark this answers is not in the document]
Comment (ramaraochavali) [reply pilot/pkg/networking/core/route/retry/retry.go:76]: it is moved up and handle both cases

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/retry/retry.go:76]: IC, nvm

File: pilot/pkg/networking/core/route/route.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/route/route.go:627]: do we have validation that checked it

Comment (ramaraochavali) [reply pilot/pkg/networking/core/route/route.go:627]: I do not think we have - but we should add IMO (atleast warn)

**Comment by ramaraochavali** [comment]
@hzxuzhonghu can you PTAL when you get chance?

**Comment by ramaraochavali** [comment]
@howardjohn @hzxuzhonghu Can you  PTAL when you get chance?

**Comment by ramaraochavali** [comment]
@hzxuzhonghu  Can you PTAL when you get chance?

**Review by howardjohn — APPROVED** [review]

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM

**Pull request merged**